Projects/Chat/Sources/Views/ChatScreenViewModel.swift

```diff
@@ -241,6 +250,8 @@ public class ChatScreenViewModel: ObservableObject {
     private func handleSendFail(for message: Message, with error: String) {
         if let index = messages.firstIndex(where: { $0.id == message.id }) {
             let newMessage = message.asFailed(with: error)
+            let store: ChatStore = globalPresentableStoreContainer.get()
+            store.send(.setFailedMessage(newMessage))
```

Comment:
This is maybe placed in the wrong place - we would append new messages again and again. We want to add it once it failed, not if it is already failed. That logic for adding should be moved a little down.
```
            switch oldMessage.status {
            case .failed:
                break
            default:
                messages[index] = newMessage
                let store: ChatStore = globalPresentableStoreContainer.get()
                store.send(.setFailedMessage(newMessage))
            }
```